vllm/model_executor/layers/logits_processor.py

```diff
@@ -76,13 +77,19 @@ def forward(
     def _get_logits(
         self,
         hidden_states: torch.Tensor,
-        lm_head: VocabParallelEmbedding,
+        lm_head: Union[VocabParallelEmbedding, LinearBase],
         embedding_bias: Optional[torch.Tensor],
     ) -> Optional[torch.Tensor]:
         # Get the logits for the next tokens.
-        logits = lm_head.linear_method.apply(lm_head,
-                                             hidden_states,
-                                             bias=embedding_bias)
+        linear_method = None
+        if isinstance(lm_head, LinearBase):
```

Comment:
Are there cases where we expect something other than `ReplicatedLinear` or `VocabParallelEmbedding`. If not, maybe it is better to make it more explicit and use `ReplicatedLinear` rather than the base class.

Reply:
It's not clear to me whether `ReplicatedLinear` should be used in TP. I've only tested using it in single GPU deployment